### PR TITLE
[Feature] Remove "displayName"

### PR DESCRIPTION
**Pull request by KaterinaWire**
## What's new in this PR?

Remove the "displayName" logic from the user.

### CODE REVIEW

File: Source/Model/Conversation/ZMConversation+DisplayName.swift [hunk 1]
```diff
@@ -80,11 +80,11 @@ public extension ZMConversation {
         if let userDefined = userDefinedName, !userDefined.isEmpty {
             return userDefined
         }
-
+        
         let activeNames: [String] = localParticipants.compactMap { (user) -> String? in
-            guard user != selfUser &&
-                !user.displayName.isEmpty else { return nil }
-            return user.displayName
+            guard let fullName = user.name,
```
Comment (typfel) [inline Source/Model/Conversation/ZMConversation+DisplayName.swift]: No need to unwrap the `user.name` in a `compactMap`

File: Source/Model/User/DisplayNameGenerator.swift [hunk 2]
```diff
@@ -50,67 +49,4 @@ private var zmLog = ZMSLog(tag: "DisplayNameGenerator")
     public func initials(for user: ZMUser) -> String? {
         return personName(for: user).initials
     }
-    
-    
-    // MARK : DisplayNames on a conversation basis
-    
-    private var currentDisplayNameMap : ConversationDisplayNameMap?
```
Comment (typfel) [inline Source/Model/User/DisplayNameGenerator.swift]: Why can't `DisplayNameGenerator.swift` be deleted completely?

File: Tests/Source/Model/User/DisplayNameGeneratorTests.swift [hunk 3]
```diff
@@ -173,19 +173,19 @@ class DisplayNameGeneratorTests : ZMBaseManagedObjectTest {
...
-        user1.name = "Rob A"
-
-        // when
-        let givenName = user1.displayName
-        
-        // then
-        XCTAssertNotNil(givenName)
-        XCTAssertEqual(givenName, "Rob")
-    }
+//    func testThatItFetchesAllUsersWhenNotFetchedYet()
```
Comment (typfel) [inline Tests/Source/Model/User/DisplayNameGeneratorTests.swift]: Commented out code,  but I would also expect to delete this whole file.

File: Tests/Source/Model/User/ZMUserDisplayNameGeneratorTest.m [hunk 4]
```diff
@@ -297,62 +297,6 @@ - (void)testThatItReturnsInitialsForUser
 
 @implementation ZMUserDisplayNameGeneratorTest (Performance)
 
-- (void)testDisplayNamePerformanceForGivenNames_FirstAccess
```
Comment (typfel) [inline Tests/Source/Model/User/ZMUserDisplayNameGeneratorTest.m]: Would expect that this whole file could be deleted.

File: Source/Model/Conversation/ZMConversation+DisplayName.swift (hunk 1 above)
Comment (KaterinaWire) [reply Source/Model/Conversation/ZMConversation+DisplayName.swift]: Fixed

File: Source/Model/User/DisplayNameGenerator.swift (hunk 2 above)
Comment (KaterinaWire) [reply Source/Model/User/DisplayNameGenerator.swift]: Deleted

File: Tests/Source/Model/User/DisplayNameGeneratorTests.swift (hunk 3 above)
Comment (KaterinaWire) [reply Tests/Source/Model/User/DisplayNameGeneratorTests.swift]: Deleted

File: Tests/Source/Model/User/ZMUserDisplayNameGeneratorTest.m (hunk 4 above)
Comment (KaterinaWire) [reply Tests/Source/Model/User/ZMUserDisplayNameGeneratorTest.m]: Deleted

**Comment by typfel** [comment]
There's also usage of the `displayName` in `wire-ios-sync-engine`

**Comment by KaterinaWire** [comment]
> There's also usage of the `displayName` in `wire-ios-sync-engine`

Yes, I created a PR

**Comment by KaterinaWire** [comment]
@zenkins release major

**Pull request merged**